internal/pkg/agent/application/upgrade/step_download.go

```diff
@@ -219,7 +219,7 @@ func (u *Upgrader) downloadOnce(
 	// All download artifacts expect a name that includes <major>.<minor.<patch>[-SNAPSHOT] so we have to
 	// make sure not to include build metadata we might have in the parsed version (for snapshots we already
 	// used that to configure the URL we download the files from)
-	path, err := downloader.Download(ctx, agentArtifact, version.VersionWithPrerelease())
+	path, err := downloader.Download(ctx, agentArtifact, version)
```

Comment:
This is the actual fix, are we able to unit or integration test this?

Reply:
I believe a mock HTTP server which is something that is really easily to create with golang is the way to test this. I agree that we need to test this to ensure that the behavior is working as expected.

I would prefer to see the test at this level as well, so its testing the whole step_download section.

Reply:
>For unit tests, given the current code structure we would have to mock an artifact api on a test HTTP server, not exactly a "unit" test

It won't be the first _not exactly a "unit" test_, besides as Blake pointed out, it's easy in Go and if you want, you could make it an integration test. 

I agree a test is important, for me it can be in any form you prefer as long as there is a test